src/buffer/out/search.cpp

```diff
@@ -344,3 +361,69 @@ std::vector<std::vector<wchar_t>> Search::s_CreateNeedleFromString(const std::ws
     }
     return cells;
 }
+
+// Routine Description:
+// - Performs a regex search instead of a regular search within the given start and end coordinates
+// - The caller is responsible for making sure the start and end coordinates are correct with respect
+//   to the direction of the search (find previous or find next)
+// Arguments:
+// - start: the start coordinate
+// - end: the end coordinate
+// Return Value:
+// - true if we found a regex match, false otherwise
+bool Search::_RegexHelper(COORD start, COORD end)
+{
+    std::wstring concatAll;
+    auto begin = start;
+    std::wsmatch match;
+
+    // To deal with text that spans multiple lines, we will first concatenate
+    // all the text into one string and find the regex in that string
+    while (begin != end)
+    {
+        concatAll += *_uiaData.GetTextBuffer().GetTextDataAt(begin);
+        _IncrementCoord(begin);
+    }
+
+    // Create the regex object and iterator
+    std::wregex regexObj{ _inputString };
+    auto matches_begin = std::wsregex_iterator(concatAll.begin(), concatAll.end(), regexObj);
+    auto matches_end = std::wsregex_iterator();
+
+    // If we found a match, get its position and length to update the selection coordinates
+    if (matches_begin != matches_end)
+    {
+        std::wsregex_iterator desired;
+        if (_direction == Direction::Forward)
+        {
+            desired = matches_begin;
+        }
+        else
+        {
+            // Regex iterators are not bidirectional, so we cannot create a reverse iterator out of one
+            // So, if we want to find the previous match, we need to manually progress the iterator
+            while (matches_begin != matches_end)
+            {
+                desired = matches_begin;
+                ++matches_begin;
+            }
+        }
+
+        const auto pos = desired->position();
+        for (auto i = 0; i < pos; ++i)
+        {
+            _IncrementCoord(start);
+        }

```

Comment:
This seems to assume that each `*_uiaData.GetTextBuffer().GetTextDataAt(begin)` in the earlier loop returned a `std::wstring_view` that has only one `wchar_t`.

Reply:
Thanks for catching that! Fixed